src/screens/payment-methods/PaymentMethods.tsx

```diff
@@ -150,15 +155,84 @@ export default class PaymentMethods extends BaseAutoRefreshScreen<Props, State>
+                  overshootLeft={false}
+                  childrenContainerStyle={style.swiperChildrenContainer}
+                  renderRightActions={() => this.renderSwipeableActions(paymentMethod, style)}>
+                  <PaymentMethodComponent paymentMethod={paymentMethod} navigation={navigation} />
+                </Swipeable>
+              )}
               refreshing={refreshing}
               manualRefresh={this.manualRefresh}
               onEndReached={this.onEndScroll}
-              emptyTitle={i18n.t('paymentMethods.noPaymentMethod')}
+              emptyTitle={I18n.t('paymentMethods.noPaymentMethod')}
             />
           </View>
         )}
       </Container>
     );
   };
+
+  private renderSwipeableActions(paymentMethod: BillingPaymentMethod, style: any) {
+    const deleteInProgress = this.state.deleteOperationsStates?.[paymentMethod.id];
+    const commonColors = Utils.getCurrentCommonColor();
+    return paymentMethod.isDefault ? null : (
+      <TouchableOpacity
+        disabled={deleteInProgress}
+        style={style.trashIconButton}
+        onPress={() => {
+          this.onPressDelete(paymentMethod);
+        }}>
+        {deleteInProgress ? (
+          <ActivityIndicator size={scale(20)} color={commonColors.textColor} />
+        ) : (
+          <Icon style={style.trashIcon} name="trash" />
+        )}
+      </TouchableOpacity>
+    );
+  }
+
+  private onPressDelete(paymentMethod: BillingPaymentMethod): void {
+    Alert.alert(
+      I18n.t('paymentMethods.deletePaymentMethodTitle'),
+      I18n.t('paymentMethods.deletePaymentMethodSubtitle', { cardBrand: paymentMethod.brand, cardLast4: paymentMethod.last4 }),
+      [
+        {
+          text: I18n.t('general.yes'),
+          onPress: () => {
+            this.deletePaymentMethod(paymentMethod.id as string);
+          }
+        },
+        { text: I18n.t('general.cancel') }
+      ]
+    );
+  }
+
+  private async deletePaymentMethod(paymentMethodID: string): Promise<void> {
+    const userID = this.centralServerProvider?.getUserInfo()?.id;
+    this.setState({ deleteOperationsStates: { ...this.state.deleteOperationsStates, [paymentMethodID]: true } });
+    try {
+      // TODO check res data success (waiting for server change)
+      const res = await this.centralServerProvider.deletePaymentMethod(userID, paymentMethodID);
+      Message.showSuccess(I18n.t('paymentMethods.deletePaymentMethodSuccess'));
+    } catch (error) {
+      // Check if HTTP?
+      if (!error.request || error.request.status !== HTTPAuthError.FORBIDDEN) {
+        Utils.handleHttpUnexpectedError(
+          this.centralServerProvider,
+          error,
+          'paymentMethods.paymentMethodUnexpectedError',
+          this.props.navigation,
+          this.refresh.bind(this)
+        );
+      }
+    } finally {
+      const deleteOperationsStates = this.state.deleteOperationsStates;
+      delete deleteOperationsStates[paymentMethodID];
```

Comment:
232 + 233, can be done in one line.

Reply:
I think the state is immutable

Reply:
> I think the state is immutable

And do you think transfering an object to a constant will make it mutable? ;-)
This should work:
```
      delete this.state.deleteOperationsStates[paymentMethodID];
      this.setState({ deleteOperationsStates: this.state.deleteOperationsStates });

```

Reply:
If the state is already modified in the first line, what would be the point of setting it to its current value after ?

Reply:
> If the state is already modified in the first line, what would be the point of setting it to its current value after ?

Triggering the render() method ;-)

Reply:
Then this.setState(this.state) would also work

Reply:
> Then this.setState(this.state) would also work

You have also another method to force the refresh but it's a weird way to update the State in the end.

Reply:
So which one do I use ?

Reply:
The one which is the most readable.